src/Serialization/Fields.jl

```diff
@@ -268,6 +268,65 @@ function load_internal_with_parent(s::DeserializerState,
     return parent(num, den)
 end
 
+################################################################################
+# RationalFunctionField
+
+@registerSerializationType(AbstractAlgebra.Generic.RationalFunctionField, "RationalFunctionField")
+
+function save_internal(s::SerializerState,
+                       RF::AbstractAlgebra.Generic.RationalFunctionField)
+    return Dict(
+        :base_ring => save_type_dispatch(s, base_ring(RF)),
+        :symbols => save_type_dispatch(s, symbols(RF))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: AbstractAlgebra.Generic.RationalFunctionField},
+                       dict::Dict)
+    R = load_unknown_type(s, dict[:base_ring])
+    symbols = load_unknown_type(s, dict[:symbols])
+
+    return RationalFunctionField(R, symbols, cached=false)[1]
+end
+
+#elements
+@registerSerializationType(AbstractAlgebra.Generic.Rat, "Rat")
+
+function save_internal(s::SerializerState, f::AbstractAlgebra.Generic.Rat)
+    frac_elem_parent = save_type_dispatch(s, parent(denominator(f)))
+    return Dict(
+        :parent => save_type_dispatch(s, parent(f)),
+        :frac_elem_parent => frac_elem_parent,
+        :den => save_type_dispatch(s, denominator(f)),
+        :num => save_type_dispatch(s, numerator(f))
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: AbstractAlgebra.Generic.Rat},
+                       dict::Dict)
+    _ = load_unknown_type(s, dict[:frac_elem_parent])
+    R = load_type_dispatch(s, AbstractAlgebra.Generic.RationalFunctionField, dict[:parent])
+    parent = base_ring(AbstractAlgebra.Generic.fraction_field(R))
```

Comment:
Huh, why does that work? I think it would be very helpful to add a comment here which explains what's going on here.

Reply:
@thofma can you explain this?
R = Q(x,y)
fraction_field(R) = FractionField(Q[x,y])
base_ring(FractionField(Q[x,y])) = Q[x,y]

Reply:
I do not understand. You are defining a `fraction_field` method in the second line?

Reply:
see update.

Reply:
There is no official way to get the underlying polynomial ring of a rational function field. So we do the detour via the `FractionField` object, of which the rational function field is build from.

Reply:
But wouldn't `parent(num(one(R))` also work and then be perhaps easier to understand?

Reply:
In either case, I vote for a (possibly tweaked) copy of the comment by @thofma to be inserted as a source comment

Reply:
yes, `parent(num(one(R))` is unfortunately the only official way. I think Antony has enough info now to decide what comment is appropriate.